packages/langium/src/workspace/documents.ts

```diff
@@ -98,6 +102,14 @@ export interface DocumentSegment {
     readonly end: number
 }
 
+/**
+ * Surrogate definition of the `TextDocuments` interface from the `vscode-languageserver` package.
+ * No implementation object is expected to be offered by `LangiumCoreServices`, but only by `LangiumLSPServices`.
+ */
+export type LangiumCoreTextDocuments = {
```

Comment:
Naming: Given that this interface isn't extended by another interface (i.e. there's no `LangiumTextDocuments` in `langium/lsp`, it probably makes sense to rename this:
```suggestion
export type TextDocumentProvider = {
```

Reply:
I get that, but I think `TextDocumentProvider` is good enough.

Reply:
I'm in favor of `TextDocumentProvider` too: it describes better what that type is about.

Reply:
done!